x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -41,14 +48,21 @@
 public class WatcherLifeCycleService implements ClusterStateListener {
 
     private static final Logger logger = LogManager.getLogger(WatcherLifeCycleService.class);
+
+    public static final Set<String> LEGACY_WATCHER_INDEX_TEMPLATES = Set.of(".watches", ".triggered_watches");
+
     private final AtomicReference<WatcherState> state = new AtomicReference<>(WatcherState.STARTED);
     private final AtomicReference<List<ShardRouting>> previousShardRoutings = new AtomicReference<>(Collections.emptyList());
     private volatile boolean shutDown = false; // indicates that the node has been shutdown and we should never start watcher after this.
+    private final ClusterService clusterService;
     private volatile WatcherService watcherService;
     private final EnumSet<WatcherState> stopStates = EnumSet.of(WatcherState.STOPPED, WatcherState.STOPPING);
+    private final AtomicBoolean legacyTemplatesDeleteInProgress = new AtomicBoolean(false);
+    private final AtomicBoolean checkForLegacyTemplates = new AtomicBoolean(true);
```

Comment:
Should checking whether a removal of templates is in progress be part of the functionality that `MetadataIndexTemplateService.removeTemplates(...)` method offers?